tests/expressions/tests.py

```diff
@@ -2333,6 +2333,34 @@ def test_resolve_output_field(self):
                 expr = Value(value)
                 self.assertIsInstance(expr.output_field, output_field_type)
 
+    def test_mixed_output_field_with_none_returns_field(self):
```

Comment:
This will probably not need to be covered with https://github.com/django/django/pull/17890, now.